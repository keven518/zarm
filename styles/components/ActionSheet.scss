$prefixCls: $prefixCls-actionsheet;

@mixin actionsheet-theme($color) {
  &,
  &:hover,
  &:focus,
  &:visited {
    text-decoration: none;
    color: $color;
  }
}

.#{$prefixCls} {
  margin: 0 r(10) r(10) r(10);

  &.shape-radius {
    .#{$prefixCls} {
      &-actions,
      &-cancel {
        overflow: hidden;
        border-radius: $radius-md;
      }
    }
  }

  &-actions,
  &-cancel {
    background: #fff;
  }

  &-cancel {
    margin-top: r(7);
  }

  &-btn {
    display: block;
    height: r(50);
    line-height: r(50);
    text-align: center;
    font-size: r(17);
    @include actionsheet-theme($theme-default);

    &:active {
      background: #f1f1f1;
    }

    &:not(:first-of-type) {
      position: relative;
      @include onepx(top);
    }

    &.theme-default {
      @include actionsheet-theme($theme-default);
    }

    &.theme-primary {
      @include actionsheet-theme($theme-primary);
    }

    &.theme-info {
      @include actionsheet-theme($theme-info);
    }

    &.theme-success {
      @include actionsheet-theme($theme-success);
    }

    &.theme-warning {
      @include actionsheet-theme($theme-warning);
    }

    &.theme-error {
      @include actionsheet-theme($theme-error);
    }
  }
}

