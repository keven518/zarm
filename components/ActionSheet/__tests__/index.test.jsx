import React from 'react';
import { render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ActionSheet from '../index';

describe('ActionSheet', () => {
  const props = {
    visible: true,
    actions: [
      {
        text: '操作一',
        onClick: jest.fn(),
      },
      {
        text: '操作二',
        onClick: jest.fn(),
      },
      {
        theme: 'error',
        text: '操作三',
        onClick: jest.fn(),
      },
    ],
    onMaskClick: jest.fn(),
  };

  it('renders correctly', () => {
    const wrapper = render(<ActionSheet {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('click cancel button', () => {
    props.onCancel = jest.fn();
    const wrapper = shallow(<ActionSheet {...props} />);
    (wrapper.find('.za-actionsheet-cancel').childAt(0)).simulate('click');
    expect(props.onCancel).toBeCalled();
  });
});
