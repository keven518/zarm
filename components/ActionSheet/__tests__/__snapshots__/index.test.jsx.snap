// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionSheet renders correctly 1`] = `
<div
  class="za-popup"
>
  <div
    class="za-popup-wrapper za-popup-wrapper-bottom"
    style="-webkit-transition-duration:200ms;transition-duration:200ms;"
  >
    <div
      class="za-actionsheet shape-radius"
    >
      <div
        class="za-actionsheet-actions"
      >
        <a
          class="za-actionsheet-btn"
        >
          操作一
        </a>
        <a
          class="za-actionsheet-btn"
        >
          操作二
        </a>
        <a
          class="za-actionsheet-btn theme-error"
        >
          操作三
        </a>
      </div>
    </div>
  </div>
  <div
    class="za-mask normal"
    style="-webkit-animation-duration:200ms;animation-duration:200ms;"
  />
</div>
`;
